src/MainWindow.vala

```diff
@@ -221,16 +226,20 @@ public class Mail.MainWindow : Hdy.ApplicationWindow {
         conversation_list.mark_unstar_selected_messages ();
     }
 
-    private void on_reply () {
-        message_list.compose.begin (Composer.Type.REPLY);
+    private void on_reply (SimpleAction action, Variant? parameter) {
+        message_list.compose.begin (Composer.Type.REPLY, parameter);
     }
 
-    private void on_reply_all () {
-        message_list.compose.begin (Composer.Type.REPLY_ALL);
+    private void on_reply_all (SimpleAction action, Variant? parameter) {
+        message_list.compose.begin (Composer.Type.REPLY_ALL, parameter);
     }
 
-    private void on_forward () {
-        message_list.compose.begin (Composer.Type.FORWARD);
+    private void on_forward (SimpleAction action, Variant? parameter) {
+        message_list.compose.begin (Composer.Type.FORWARD, parameter);
+    }
```

Comment:
Since we get the action as an argument, we could just have a single function:
```
    private void action_compose (SimpleAction action, Variant? parameter) {
        switch (action.name) {
            case ACTION_REPLY:
                message_list.compose.begin (Composer.Type.REPLY, parameter);
                break;
            case ACTION_REPLY_ALL:
                message_list.compose.begin (Composer.Type.REPLY_ALL, parameter);
                break;
            case ACTION_FORWARD:
                message_list.compose.begin (Composer.Type.FORWARD, parameter);
                break;
        }
    }
```